infra/RepoManager.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Class to manage a git repository via python.
+
+This class is to be used to implement git commands over
+a python API and manage the current state of the git repo.
+
+  Typical usage example:
+
+    r_man =  RepoManager('https://github.com/google/oss-fuzz.git')
+    r_man.checkout('5668cc422c2c92d38a370545d3591039fb5bb8d4')
+"""
+import logging
+import os
+import shutil
+import subprocess
+
+
+class RepoManagerError(Exception):
+  """Class to describe the exceptions in RepoManager."""
+
+  def __init__(self, message):
+    """ Init the exception class.
+
+    Args:
+      message: the message to propagate to user
+    """
+    super().__init__(message)
+
+
+class RepoManager(object):
+  """Class to manage git repos from python.
+
+  Attributes:
+    repo_url: The location of the git repo
+    base_dir: The location of where the repo clone is stored locally
+    repo_name: The name of the github project
+    repo_dir: The location of the main repo
+  """
+
+  def __init__(self, repo_url, base_dir):
```

Comment:
nit: repo_dir is probably a clearer name. base is a little ambiguous in that it may imply that a subdir is created in it for the actual repo.

Reply:
A sub directory is created in base directory on the git clone call.

Reply:
Ah, I misread line 68, and thought you were passing the base_dir as part of "git clone URL base_dir". 

That said, I believe getting the client to pass the full repo path is a better API, i.e. `git clone URL repo_dir`. That way, the caller does not have to check/guess where the repo was actually checked out to.

Reply:
running `git clone URL` being run from base_dir would have the same effect?

Reply:
Yes, just a nit in that the caller can provide the full repo path instead, which may be more intuitive than passing a `base_dir`. In any case, feel free to ignore -- thinking about this a bit more I can see why both ways are reasonable, so any further discussion is probably just bikeshedding :)